rbx_reflection/src/database.rs

```diff
@@ -30,6 +30,23 @@ pub struct ReflectionDatabase<'a> {
     pub enums: HashMap<Cow<'a, str>, EnumDescriptor<'a>>,
 }
 
+pub struct SuperClassIter<'a> {
+    database: &'a ReflectionDatabase<'a>,
+    descriptor: Option<&'a ClassDescriptor<'a>>,
+}
+impl<'a> Iterator for SuperClassIter<'a> {
+    type Item = &'a ClassDescriptor<'a>;
+    fn next(&mut self) -> Option<Self::Item> {
+        let next_descriptor = self.descriptor.and_then(|descriptor| {
+            descriptor
+                .superclass
+                .as_ref()
+                .and_then(|class_name| self.database.classes.get(class_name))
+        });
+        core::mem::replace(&mut self.descriptor, next_descriptor)
```

Comment:
```suggestion
        let superclass = self.descriptor?.superclass.as_ref()?;
        let next_descriptor = self.database.classes.get(superclass)?;
        self.descriptor.replace(next_descriptor)
```
You can just use `?` for this since we want to exit if these are `None` anyway. Also, `Option::replace` is just a better option for this.

Reply:
This is great!